components/AudioPlayer.tsx

```diff
@@ -0,0 +1,194 @@
+            onClick={() => {
+              setLoadingDownload(true);
+              setInterval(() => setLoadingDownload(false), 5000);
+            }}
+          >
+            <i
+              className={`fa-solid fa-${
+                loadingDownload ? 'spinner animate-spin' : 'cloud-arrow-down'
+              }`}
+            />
+          </a>
+        </div>
+        <div className="flex-grow flex items-center shrink-0 space-x-2 order-first sm:order-none w-full sm:w-auto mb-2 sm:mb-0">
+          <span className="w-16 text-slate-400">{formatDuration(currentTime)}</span>
+          <div className="relative flex-grow -mt-1.5">
+            <input
+              ref={progressBar}
+              type="range"
+              className={styles.progressBar}
+              value={currentTime}
+              onChange={() => {
+                if (audioPlayer.current) {
+                  audioPlayer.current.currentTime = Number(progressBar.current?.value);
+                  setCurrentTime(Number(progressBar.current?.value));
+                }
+              }}
+            />
+            <div
+              className="absolute h-[6px] left-0 top-0 rounded-l-full z-10 bg-sky-500"
+              style={{
+                width: `calc(${(100 * currentTime) / duration}% - ${
+                  (16 * currentTime) / duration
+                }px + 1px)`,
+              }}
+            />
+          </div>
+          <span className="w-16 text-slate-400 flex justify-end">
+            {formatDuration(duration)}
+          </span>
+        </div>
+        <div className="shrink-0 flex justify-end sm:justify-center items-center flex-grow sm:flex-grow-0">
+          <button
+            className="hover:bg-slate-200/50 dark:hover:bg-slate-700 w-8 h-8 flex justify-center items-center rounded-full text-slate-400 dark:text-slate-500 dark:hover:text-slate-400  transition duration-100"
+            onClick={() => {
+              if (audioPlayer.current) {
+                const prevValue = muted;
+                setMuted(!prevValue);
+                audioPlayer.current.volume = prevValue ? 1 : 0;
+              }
+            }}
+          >
+            {muted ? (
+              <SpeakerXMarkIcon className="w-5" />
+            ) : (
+              <SpeakerWaveIcon className="w-5" />
+            )}
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default AudioPlayer;
+
+function formatDuration(seconds: number): string {
+  if (isNaN(seconds)) {
+    return `00:00`;
+  }
+  const hourPart = Math.floor(seconds / 3600);
+  const minutePart = Math.floor((seconds - hourPart * 3600) / 60);
+  const secondPart = seconds % 60;
+  return `${hourPart ? `${hourPart}:` : ''}${addZero(minutePart)}:${addZero(secondPart)}`;
+}
+
+function addZero(num: number): string {
```

Comment:
or you could use `string.padStart`: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart